specification/cognitiveservices/OpenAI.Inference/models/completions/extensions_onyourdata_vector_search.tsp

```diff
@@ -0,0 +1,63 @@
+import "@typespec/versioning";
+import "./extensions_onyourdata_authentication.tsp";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@doc("Embedding dependency types for vector search.")
+enum OnYourDataEmbeddingDependencyType {
+  @doc("Azure OpenAI Embedding endpoint URL.")
+  endpoint: "Endpoint",
+
+  @doc("Azure OpenAI Embedding deployment name.")
+  deploymentName: "DeploymentName",
+
+  @doc("The embedding model id defined in the search service. Currently only supported by Elasticsearch.")
+  modelId: "ModelId",
+}
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@discriminator("type")
+@doc("Embedding dependency for vector search.")
+model OnYourDataEmbeddingDependency {
+  @doc("Embedding dependency types for vector search.")
+  type: OnYourDataEmbeddingDependencyType;
+}
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@doc("The embedding dependency based on endpoint URL")
+model OnYourDataEmbeddingEndpointDependency
```

Comment:
nit: It could be valuable to describe what an "embedding dependency" is.

Reply:
Embedding dependency tells the service how to calculate the embedding of the intent to retrieve the data using vector search. I just look at how Azure AI Search name it - they call it vectorizer, seems to be a better name than embedding dependency, thoughts? https://learn.microsoft.com/en-us/azure/search/vector-search-how-to-configure-vectorizer#define-a-vectorizer